scaling/provider/spec.go

```diff
@@ -0,0 +1,9 @@
+package provider
+
+type Interface interface {
+	AddWorker() error
+	NumMasters() (int, error)
+	NumWorkers() (int, error)
+	RemoveWorker() error
+	WaitForNodesUp(num int) error
+}
```

Comment:
This is pretty cool. We only have to figure how we have to implement these pieces for Azure and KVM and have the e2e test for scaling sorted.

Reply:
Yes, the interface here is nice.

Naming so ¯\_(ツ)_/¯ and I know it's in the framework but I wondered if this could be `WaitForNodes` or `WaitForReadyNodes`?

Reply:
`WaitForNodes` it is.